app/main.py

```diff
@@ -1,2 +1,54 @@
+from typing import Hashable, Any
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+
+    def __repr__(self) -> str:
+        return f"{self.key}: {self.value}"
+
+
 class Dictionary:
-    pass
+    NEW_LEVEL = 2 / 3
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+        for node in old_hash_table:
+            if node is not None:
+                self[node.key] = node.value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % self.capacity
+        while self.hash_table[index] is not None:
```

Comment:
You are using this algorithm twice inside your class. 
May be it would be better to create separate method and just use it instead repeating whole code?